test/testsuite.at

```diff
@@ -95,11 +95,11 @@ fi
 ]
 m4_divert_pop([PREPARE_TESTS])
 
-AT_BANNER(Read [MAT version 4 (Little Endian)])
+AT_BANNER([Read MAT version 4 (Little Endian)])
 m4_include([mat4_read_le.at])
 m4_include([mat4_readslab_le.at])
 
-AT_BANNER(Read [MAT version 4 (Big Endian)])
+AT_BANNER([Read MAT version 4 (Big Endian)])
```

Comment:
These changes are independent of the Cmake stuff and should be an extra PR - if you like.

Reply:
Why is it necessary to move all at files? I'd prefer to keep it as is.

Reply:
At the beginning, I was considering creating a cmake testsuite mimicking the autotools one. That would have made a lot of supplementary files and the directory would have been cluttered. So I put all the testsuite files (besides the main one) in their own directory. Then, I knew it would be difficult to maintain two testsuites in parallel (and I saw your comment of the matio-pubconf.h file), so I went the route of trying to parse the current testsuite (which is somewhat fragile as I'm doing pattern matching, so any change in the testsuite might create problems but seems to work). 

Yet, I find the organization of the testsuite in a subdir cleaner...

At this point, I'm first trying to have everything working and in place with at least one travis build working (have not found appveyor configuration). If you insist on keeping the files in their original place, I will do it, but I think the reorganization is somewhat better...

Reply:
For the other PR, I afraid of conflicts. But if you prefer, I  surely can do that.

Reply:
The Appveyor config is only in the cloud. It basically calls msbuild on the VS2010 solution.

This already looks like a monster PR. Keeping some independent changes in its own PR might help for clarity.

Reply:
> If you insist on keeping the files in their original place, I will do it, but I think the reorganization is somewhat better...

Not sure what to prefer. But as maintainer I was rather conservative here and tried to avoid any structural changes to the original code.

Does https://cmake.org/Wiki/CMakeEmulateMakeCheck give some clue?